specs/predeploys.md

```diff
@@ -0,0 +1,657 @@
+# Predeploys
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [OVM\_L2ToL1MessagePasser](#ovm%5C_l2tol1messagepasser)
+- [OVM\_DeployerWhitelist](#ovm%5C_deployerwhitelist)
+- [OVM\_ETH](#ovm%5C_eth)
+- [WETH9](#weth9)
+- [L2CrossDomainMessenger](#l2crossdomainmessenger)
+- [L2StandardBridge](#l2standardbridge)
+- [SequencerFeeVault](#sequencerfeevault)
+- [L2StandardTokenFactory](#l2standardtokenfactory)
+- [L1BlockNumber](#l1blocknumber)
+- [OVM\_GasPriceOracle](#ovm%5C_gaspriceoracle)
+- [L1Attributes](#l1attributes)
+- [Withdrawer](#withdrawer)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+Predeployed smart contracts exist on Optimism at predetermined addresses in
+the genesis state. They are similar to precompiles but instead run directly
+in the EVM instead of running native code outside of the EVM.
```

Comment:
Once we know the height at which we switch to bedrock, we could add that to this document. Maybe it could go into a file dedicated to backwards compatibility?